src/Features/Core/Portable/CodeRefactorings/AbstractRefactoringHelpersService.cs

```diff
@@ -0,0 +1,176 @@
+                ? tokenOnSelection
+                : tokenOnSelection.GetPreviousToken();
+
+            var tokenToLeft = (tokenPreSelection.Span.End == selectionStripped.Start)
+                ? tokenPreSelection
+                : default;
+
+            if (tokenToLeft != default)
+            {
+                var leftNode = tokenToLeft.Parent;
+                do
+                {
+                    // consider either a Node that is a parent of touched Token (selection can be within) or ancestor Node of such Token whose span ends on selection
+                    var wantedNode = TryGetAcceptedNodeOrExtracted(leftNode, predicate, extractNode, syntaxFacts);
+                    if (wantedNode != default)
+                    {
+                        return wantedNode;
+                    }
+
+                    leftNode = leftNode?.Parent;
+                }
+                while (leftNode != null && leftNode.Span.End == selection.Start);
+            }
+
+            // nothing found
+            return default;
+
+            static SyntaxNode TryGetAcceptedNodeOrExtracted(SyntaxNode node, Predicate<SyntaxNode> predicate, Func<SyntaxNode, ISyntaxFactsService, SyntaxNode> extractNode, ISyntaxFactsService syntaxFacts)
+            {
+                if (node == default)
+                {
+                    return default;
+                }
+
+                if (predicate(node))
+                {
+                    return node;
+                }
+
+                var extrNode = extractNode(node, syntaxFacts);
+                if (extrNode != default && predicate(extrNode))
+                {
+                    return extrNode;
+                }
+
+                return default;
+            }
+        }
+
+        public virtual SyntaxNode DefaultNodeExtractor(SyntaxNode node, ISyntaxFactsService syntaxFacts)
+        {
+            // var a = b;
+            // -> b
+            if (syntaxFacts.IsLocalDeclarationStatement(node))
+            {
+                var variables = syntaxFacts.GetVariablesOfLocalDeclarationStatement(node);
+                if (variables.Count == 1)
+                {
+                    var declaredVariable = variables.First();
+                    var initializer = syntaxFacts.GetInitializerOfVariableDeclarator(declaredVariable);
+
+                    if (initializer != default)
+                    {
+                        var value = syntaxFacts.GetValueOfEqualsValueClause(initializer);
+                        if (value != default)
+                        {
+                            return value;
+                        }
+                    }
+                }
+            }
+
+            // a = b;
+            // -> b
+            if (syntaxFacts.IsSimpleAssignmentStatement(node))
+            {
+                syntaxFacts.GetPartsOfAssignmentExpressionOrStatement(node, out _, out _, out var rightSide);
+                if (rightSide != default)
```

Comment:
fyi, RHS can never be default for an assignment expr.  so you can remove this check.